.github/workflows/build.yml

```diff
@@ -28,5 +28,6 @@ jobs:
           key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
           restore-keys: |
             ${{ runner.os }}-gradle-
-      - run: |
-          ./gradlew build --no-daemon
+      - run: ./gradlew build
+      - env:
+          VERSION: "0.0.0"
```

Comment:
There is still the question: Why you want to do this ?

Reply:
because of way how our code has been checked out. There's no reason to checkout everything. and VERSION is a way to skip git to describe shallow checkout.

Reply:
Could you explain me the logic we have on master branch with this `VERSION` env variable? I quite don't understand it to continue to support it

Reply:
`3.0.1-15-g260d339`

tells you, that you 15 commits behind last tag with exact sha1 to identify it any time.

You don't need this at all
```
- env:
          VERSION: "0.0.0"
```
This is a micro mini repo, and I see no need for a shallow clone.

Reply:
We could remove this `VERSION` from build script as well

Reply:
please, follow commands checkout plugin does and you'll get an error that git can't describe anything because it's a shallow checkout by a commit number without any tags

Reply:
Groovy shell execution just ignored exit code completely, Kotlin doesn't

Reply:
If you really what to have shallow clone, what's about this ?

```
fun getTag(): String {
    try {
        val gitVersion = shellExec("git describe --tags --dirty --long").trim()
        val parts = gitVersion.split("-")
        ....
    } catch (eg: Exception) {
        retrun "0.0.0"
    }
}
```

No `VERSION` variable at all

Reply:
No I see you shallow clone pain

Reply:
I don't want to hide `shellExec` exception when we should obtain tag from git but we don't.

Reply:
How does the exception looks like ? Maybe transform it anyhow `0.0.0-no-git-version` or similar ?

Reply:
I restored old `VERSION` logic to make it compatible with `release` action. Otherwise return `0.0.0-no-git-version` as you suggested. Execution of `git` was removed completely as redundant

Reply:
@hannesa2 please, review new logic